ui/src/external/monaco.flux.server.ts

```diff
@@ -33,38 +36,76 @@ import {
   WorkspaceEdit,
   Location,
   SymbolInformation,
+  TextEdit,
 } from 'monaco-languageclient/lib/services'
 import {Server} from '@influxdata/flux-lsp-browser'
 
 type BucketCallback = () => Promise<string[]>
+type MeasurementsCallback = (bucket: string) => Promise<string[]>
 
 export interface WASMServer extends Server {
   register_buckets_callback: (BucketCallback) => void
+  register_measurements_callback: (MeasurementsCallback) => void
 }
 
 import {format_from_js_file} from '@influxdata/flux'
 
+// NOTE: parses table then select measurements from the _value column
+const parseMeasurementsResponse = response => {
+  const data = parse(response.csv) || [{data: [{}]}]
+  return data.slice(1).map(r => r[3])
+}
+
+const queryMeasurements = async (orgID, bucket) => {
+  if (!this.orgID || this.orgID === '') {
+    throw new Error('no org is provided')
+  }
+
+  const query = `import "influxdata/influxdb/v1"
+      v1.measurements(bucket:"${bucket}")`
+
+  const raw = await runQuery(orgID, query).promise
+  if (raw.type !== 'SUCCESS') {
+    throw new Error('failed to get measurements')
+  }
+}
+
 export class LSPServer {
   private server: WASMServer
   private messageID: number = 0
   private buckets: string[] = []
+  private orgID: string = ''
   private documentVersions: {[key: string]: number} = {}
   public store: Store<AppState & LocalStorage>
 
   constructor(server: WASMServer, reduxStore = store) {
     this.server = server
     this.server.register_buckets_callback(this.getBuckets)
+    this.server.register_measurements_callback(this.getMeasurements)
     this.store = reduxStore
   }
 
   getBuckets = () => {
     return Promise.resolve(this.buckets)
   }
 
+  getMeasurements = async (bucket: string) => {
+    try {
+      const response = await queryMeasurements(this.orgID, bucket)
+      return parseMeasurementsResponse(response)
+    } catch (e) {
+      return []
+    }
+  }
+
   updateBuckets(buckets: string[]) {
     this.buckets = buckets
   }
 
+  setOrg(orgID: string) {
```

Comment:
shouldn't this send a message to the LSP in case we need to update the buckets or measurements lists? or am i imagining that the LSP has more state than it needs, and if so is a follow up task to handle caching of the measurements query on the monitoring team side?

Reply:
The LSP has no state about the selected bucket or measurements and instead does smart detection for when it needs a updated list. 

- if you use a bucket parameter in a function call
- if you use a measurement parameter in a function call
- if you have a `==` with `r._measurement` on the left side

no state required on the LSP side (other than the cache of what the script looks like)